galata/src/benchmarkReporter.ts

```diff
@@ -178,17 +178,35 @@ class BenchmarkReporter implements Reporter {
    *   comparison: Logic of test comparisons: 'snapshot' or 'project'
    *    - 'snapshot': (default) This will compare the 'actual' result with the 'expected' one
    *    - 'project': This will compare the different project
+   *   graphConfigFactory: Function to create  VegaLite configuration from test records.
+   *   mdTableFactory: Function to create  markdown table from test records.
    */
   constructor(
-    options: { outputFile?: string; comparison?: 'snapshot' | 'project' } = {}
+    options: {
+      outputFile?: string;
+      comparison?: 'snapshot' | 'project';
+      graphConfigFactory?: (
+        allData: Array<IReportRecord>,
+        comparison: 'snapshot' | 'project'
+      ) => Record<string, any>;
```

Comment:
Let's clarify that the output is expected to be a _VegaList_ configuration.

```suggestion
      vegaLiteConfigFactory?: (
        allData: Array<IReportRecord>,
        comparison: 'snapshot' | 'project'
      ) => JSONObject;
```